### PR TITLE
Add IAM policy permissions to s3 tech docs docs

### PR DESCRIPTION
**Pull request by Xantier**
List minimal permissions needed to write and read tech docs on AWS environment

Signed-off-by: Jussi Hallila <jussi@hallila.com>

## Hey, I just made a Pull Request!
- [n/a] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [n/a] Tests for new functionality and regression tests for bug fixes
- [n/a] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: docs/features/techdocs/using-cloud-storage.md [hunk 1]
```diff
@@ -135,7 +135,21 @@ techdocs:
       bucketName: 'name-of-techdocs-storage-bucket'
 ```
 
-**3a. (Recommended) Setup authentication the AWS way, using environment
+**3. Create minimal AWS IAM policies to manage tech docs**
```
Comment (emmaindal) [inline docs/features/techdocs/using-cloud-storage.md:138]:
```suggestion
**3. Create minimal AWS IAM policies to manage TechDocs**
```

File: docs/features/techdocs/using-cloud-storage.md [hunk 2]
```diff
@@ -135,7 +135,21 @@ techdocs:
...
+
+To _write_ tech docs into the S3 bucket the IAM policy needs to have at a
```
Comment (emmaindal) [inline docs/features/techdocs/using-cloud-storage.md:140]:
```suggestion
To _write_ TechDocs into the S3 bucket the IAM policy needs to have at a
```

File: docs/features/techdocs/using-cloud-storage.md [hunk 3]
```diff
@@ -135,7 +135,21 @@ techdocs:
...
+minimum permissions to:
+
+- `s3:ListBucket` to retrieve bucket metadata
+- `s3:PutObject` to upload files to the bucket
+
+To _read_ tech docs from the S3 bucket the IAM policy needs to have at a minimum
```
Comment (emmaindal) [inline docs/features/techdocs/using-cloud-storage.md:146]:
```suggestion
To _read_ TechDocs from the S3 bucket the IAM policy needs to have at a minimum
```

**Review by emmaindal — COMMENTED** [review]
Nice, thank you! Just some consistency in the use of `TechDocs`.

File: docs/features/techdocs/using-cloud-storage.md (hunk 1 above)
Comment (Xantier) [reply docs/features/techdocs/using-cloud-storage.md]: Thanks!

**Comment by OrkoHunter** [comment]
Sadly DCO makes it trickier to accept commit suggestions on the web. Although when accepting suggestions, we can manually added `Signed-off-by` in the commit message.

**Comment by Xantier** [comment]
Force pushed because of DCO...

**Review by emmaindal — APPROVED** [review]

**Pull request merged**